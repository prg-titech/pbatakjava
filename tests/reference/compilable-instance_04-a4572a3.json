{
    "basename": "compilable-instance_04-a4572a3",
    "cmd": "{infile}",
    "infile": "tests/instance_04/build.sh",
    "infile_hash": "09977858e1dc340ad8e47fdee8a723605a99977e6468e5898531aafa",
    "stdout": "compilable-instance_04-a4572a3.stdout",
    "stdout_hash": "0bed911b81e5f01ce507b23ae86bb43b8f6229f1ac6b2fb35cf0131d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}