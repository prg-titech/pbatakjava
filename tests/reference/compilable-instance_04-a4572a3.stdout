
public interface Color {
}

public interface Color_Factory {
  public Color make();
}

class Color_v1 implements Color {
  public Color_v1() {
  }
}

public class Color_v1_Factory implements Color_Factory {
  public Color make() {
    return new Color_v1();
  }
}

public interface Point {
}

public interface Point_Factory {
  public Point make(boolean c);

  public Point make(Color_Factory V, Color c);
}

class Point_v1 implements Point {
  private boolean c;

  public Point_v1(boolean c) {
    this.c = c;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(boolean c) {
    return new Point_v1(c);
  }

  public Point make(Color_Factory V, Color c) {
    return null;
  }
}

class Color_v2 implements Color {
  public Color_v2() {
  }
}

public class Color_v2_Factory implements Color_Factory {
  public Color make() {
    return new Color_v2();
  }
}

class Point_v2 implements Point {
  public Color_Factory V;

  private Color c;

  public Point_v2(Color_Factory V, Color c) {
    this.V = V;
    this.c = c;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(Color_Factory V, Color c) {
    return new Point_v2(V, c);
  }

  public Point make(boolean c) {
    return null;
  }
}

public interface Line {
  public Line update(Point_Factory Y, Point p);

  public void update(Color_Factory Y);
}

public interface Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2);
}

class Line_v1 implements Line {
  public Point_Factory V;

  public Point_Factory W;

  public Point p1;

  public Point p2;

  public Line_v1(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    this.V = V;
    this.W = W;
    this.p1 = p1;
    this.p2 = p2;
  }

  public Line update(Point_Factory Y, Point p) {
    return new Line_v1(V, Y, p1, p);
  }

  public void update(Color_Factory Y) {
    this.p1 = V.make(Y, new Color_v1());
  }
}

public class Line_v1_Factory implements Line_Factory {
  public Line make(Point_Factory V, Point_Factory W, Point p1, Point p2) {
    return new Line_v1(V, W, p1, p2);
  }
}
